var/spack/repos/builtin/packages/intel-mpi/package.py

```diff
@@ -132,8 +192,29 @@ def setup_environment(self, spack_env, run_env):
         # TODO: At some point we should split setup_environment into
         # setup_build_environment and setup_run_environment to get around
         # this problem.
-        mpivars = os.path.join(
+        ver_dir = os.path.join(
```

Comment:
@adamjstewart @alalazo this will make impossible using `bindir.mpiicc` to join the path:
```
Error: AttributeError: 'str' object has no attribute 'mpiicc'
```
Whatever prefix class is, can we add a dynamic folder name and keep its class?

Reply:
I worked around this issue by doing `Prefix(some_string)`. Not sure if it's possible to extend [Prefix](https://github.com/spack/spack/blob/develop/lib/spack/spack/util/prefix.py) class to have something like
```
self.prefix.join('a_{0}'.format(1)).lib64
```

Reply:
see https://github.com/spack/spack/pull/8329